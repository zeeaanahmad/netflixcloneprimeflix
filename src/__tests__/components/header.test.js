import React from "react";
import { render, fireEvent, getByTestId } from "@testing-library/react";
import { Header } from "../../components";

jest.mock("react-router-dom", () => ({
  Link: "Link",
}));

describe("<Header/>", () => {
  it("renders the basic <Header/> component with a background", () => {
    const { container, getByText, getByTestId } = render(
      <Header>
        <Header.Frame>
          <Header.Logo src="/logo.svg" alt="logo" />
          <Header.TextLink active="true">Hello I'm a link!</Header.TextLink>
        </Header.Frame>
      </Header>
    );
    expect(getByText("Hello I'm a link!")).toBeTruthy();
    expect(getByTestId("header-bg")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the basic <Header/> component without a background", () => {
    const { container, getByText, queryByTestId } = render(
      <Header bg={false}>
        <Header.Frame>
          <Header.Logo src="/logo.svg" alt="logo" />
          <Header.ButtonLink>Sign In</Header.ButtonLink>
          <Header.TextLink active="false">Hello I'm a link!</Header.TextLink>
        </Header.Frame>
      </Header>
    );
    expect(getByText("Hello I'm a link!")).toBeTruthy();
    expect(queryByTestId("header-bg")).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the full <Header/> with a background", () => {
    const { container, getByText, getByTestId } = render(
      <Header src="joker1" dontShowBg>
        <Header.Frame>
          <Header.Group>
            <Header.Logo src="/images/logo.svg" alt="site logo" />
            <Header.TextLink active={false} onClick={() => {}}>
              Shows
            </Header.TextLink>
            <Header.TextLink active onClick={() => {}}>
              Movies
            </Header.TextLink>
          </Header.Group>
          <Header.Group>
            <Header.Search searchTerm="Joker" setSearchTerm={() => {}} />
            <Header.Profile>
              <Header.Picture src="/images/test.png" />
              <Header.Dropdown>
                <Header.Group>
                  <Header.Picture src="/images/test.png" />
                  <Header.TextLink>Test</Header.TextLink>
                </Header.Group>
                <Header.Group>
                  <Header.TextLink onClick={() => {}}>Sign out</Header.TextLink>
                </Header.Group>
              </Header.Dropdown>
            </Header.Profile>
          </Header.Group>
        </Header.Frame>
        <Header.Feature>
          <Header.FeatureTitle>Watch Joker Now</Header.FeatureTitle>
          <Header.Text>Forever alone in a crowd...</Header.Text>
          <Header.PlayButton>Play</Header.PlayButton>
        </Header.Feature>
      </Header>
    );
    expect(getByTestId("search-input")).toBeTruthy();
    expect(getByTestId("search-input").value).toBe("Joker");
    fireEvent.change(getByTestId("search-input"), {
      target: { value: "Simpsons" },
    });
    fireEvent.click(getByTestId("search-click"));
    expect(getByText("Shows")).toBeTruthy();
    expect(getByText("Movies")).toBeTruthy();
    expect(getByText("Test")).toBeTruthy();
    expect(getByText("Watch Joker Now")).toBeTruthy();
    expect(getByText("Sign out")).toBeTruthy();
    expect(getByText("Forever alone in a crowd...")).toBeTruthy();
    expect(getByText("Play")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
