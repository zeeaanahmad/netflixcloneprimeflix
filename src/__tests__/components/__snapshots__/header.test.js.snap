// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> renders the basic <Header/> component with a background 1`] = `
<div
  class="header__Background-sc-1evkybc-2 jGNIsG"
  data-testid="header-bg"
>
  <div
    class="header__Container-sc-1evkybc-0 shHWo"
  >
    <link>
      <img
        alt="logo"
        class="header__Logo-sc-1evkybc-3 dzmwll"
        src="/logo.svg"
      />
    </link>
    <p
      class="header__Link-sc-1evkybc-8 dYCvQx"
    >
      Hello I'm a link!
    </p>
  </div>
</div>
`;

exports[`<Header/> renders the basic <Header/> component without a background 1`] = `
<div
  class="header__Container-sc-1evkybc-0 shHWo"
>
  <link>
    <img
      alt="logo"
      class="header__Logo-sc-1evkybc-3 dzmwll"
      src="/logo.svg"
    />
  </link>
  <link
    class="header__ButtonLink-sc-1evkybc-4 bXFBVN"
  >
    Sign In
  </link>
  <p
    class="header__Link-sc-1evkybc-8 ejqstp"
  >
    Hello I'm a link!
  </p>
</div>
`;

exports[`<Header/> renders the full <Header/> with a background 1`] = `
<div
  class="header__Background-sc-1evkybc-2 fSCUiz"
  data-testid="header-bg"
  src="joker1"
>
  <div
    class="header__Container-sc-1evkybc-0 shHWo"
  >
    <div
      class="header__Group-sc-1evkybc-9 hFDBvc"
    >
      <link>
        <img
          alt="site logo"
          class="header__Logo-sc-1evkybc-3 dzmwll"
          src="/images/logo.svg"
        />
      </link>
      <p
        class="header__Link-sc-1evkybc-8 ejqstp"
      >
        Shows
      </p>
      <p
        class="header__Link-sc-1evkybc-8 ejqstp"
      >
        Movies
      </p>
    </div>
    <div
      class="header__Group-sc-1evkybc-9 hFDBvc"
    >
      <div
        class="header__Search-sc-1evkybc-13 tMvwX"
      >
        <button
          class="header__SearchIcon-sc-1evkybc-14 ijFFkA"
          data-testid="search-click"
        >
          <img
            alt="search"
            src="images/icons/search.png"
          />
        </button>
        <input
          class="header__SearchInput-sc-1evkybc-15 gayERW"
          data-testid="search-input"
          placeholder="Search movies and shows"
          value="Joker"
        />
      </div>
      <div
        class="header__Profile-sc-1evkybc-12 fGZQAS"
      >
        <button
          class="header__Picture-sc-1evkybc-10 jzEVPF"
          src="/images/users//images/test.png.png"
        />
        <div
          class="header__Dropdown-sc-1evkybc-11 kGqHZs"
        >
          <div
            class="header__Group-sc-1evkybc-9 hFDBvc"
          >
            <button
              class="header__Picture-sc-1evkybc-10 jzEVPF"
              src="/images/users//images/test.png.png"
            />
            <p
              class="header__Link-sc-1evkybc-8 ejqstp"
            >
              Test
            </p>
          </div>
          <div
            class="header__Group-sc-1evkybc-9 hFDBvc"
          >
            <p
              class="header__Link-sc-1evkybc-8 ejqstp"
            >
              Sign out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="header__Container-sc-1evkybc-0 header__Feature-sc-1evkybc-5 shHWo QNhqu"
  >
    <h2
      class="header__FeatureTitle-sc-1evkybc-7 ccVEYg"
    >
      Watch Joker Now
    </h2>
    <p
      class="header__Text-sc-1evkybc-6 hxrMGU"
    >
      Forever alone in a crowd...
    </p>
    <button
      class="header__PlayButton-sc-1evkybc-16 hsjOdt"
    >
      Play
    </button>
  </div>
</div>
`;
